### PR TITLE
[WIP] Enable Inventory to fetch multiple VCs per vROps

### LINKED ISSUE #204: Inventory can only handle one vCenter at a time
hi, Team

i am running a issue with the exporter.. i would like seek your help here.
[Issue description]
I have a vrops with multiple vcenters , however, when every query against that inventory url with vcenters, it  always replies with a different vcenter, not all of them. how can the exporter query all the vcenters ?

i attached the test output here. FYI.   
[update-consolidate.log](https://github.com/sapcc/vrops-exporter/files/8283360/update-consolidate.log)

pls help on this issue. thanks.

thanks
Jeoff

### PR DESCRIPTION
**Pull request by viennaa**
fixes #204 

Introducing inline object tree for easier navigation
removing parent allocation loop as this is easier now; still loopelicous though.

### CODE REVIEW

**Review by richardtief — COMMENTED** [review]
A good start to improve the underlying data structure. But to enable inventory to support multiple vCenters per vROps, we can use the existing structure for now.

File: InventoryBuilder.py [hunk 1]
```diff
@@ -259,92 +263,89 @@ def query_inventory_permanent(self):
...
+                hosts                         = list(),
+                datastores                    = list(),
+                vms                           = list(),
+                distributed_virtual_switches  = list(),
+                nsxt_adapter                  = list(),
+                nsxt_mgmt_cluster             = list(),
+                nsxt_mgmt_nodes               = list(),
+                nsxt_mgmt_service             = list(),
+                nsxt_transport_nodes          = list(),
+                nsxt_logical_switches         = list(),
```
[the remark this answers is not in the document]
Comment (viennaa) [reply InventoryBuilder.py:330]: Yes, this was from the initial idea only.

File: InventoryBuilder.py [hunk 2]
```diff
@@ -456,165 +457,157 @@ def create_sddc_health_objects(self, vrops, target: str, token: str, inventory_c
...
-                            }
-            self.amount_resources[target]['vms'] = len(tree[target])
+        for vrops in self.vrops_list:
+            for vcenter in vrops.vc:
+                vc = vrops.vc[vcenter]
+                tree[vc.vcenter.name] = dict()
+                for vm in vc.vms:
+                    ho = [ho for ho in vc.hosts if ho.uuid == vm.parent][0]
+                    cl = [cl for cl in vc.cluster if cl.uuid == ho.parent][0]
+                    dc = [dc for dc in vc.datacenter if dc.uuid == cl.parent][0]
```
[the remark this answers is not in the document]
Comment (viennaa) [reply InventoryBuilder.py:575]: And it certainly is, thought of readding the parent information here earlier too to and leverage the actual object for it

**Comment by viennaa** [comment]
Fixed with #211 already

**Pull request closed without merging**